### DISCUSSION TITLE
Error Top language

### DISCUSSION THREAD

**Discussion by Irissuu** [community/community#127021] [Other Feature Feedback, Questions, & Ideas] [answered]
### Body

Hi guys, why is it saying in my top languages ​​no data language? It was normal and out of nowhere it became like this 
![Captura de tela 2024-06-01 192436](https://github.com/community/community/assets/161527170/310dbc48-b51e-4ab4-a175-f7ff74bd28f2)
![Captura de tela 2024-06-01 192423](https://github.com/community/community/assets/161527170/eac5e027-d5bf-45c8-a39b-095acabe4727) 
(below I just deleted some < for a better view of the code)
a href="https://github.com/irissuu/github-readme-stats">
 img align="left" src="https://github-readme-stats.vercel.app/api/top-langs/?username=irissuu&layout=compact&bg_color=ff00&title_color=F9A0AF&hide_border=True&include_all_commits=true&count_private=true" width="40%" alt="Github Stats"/>
/a>
<a href="https://github.com/irissuu/github-readme-stats">
 <img align="left" src="https://github-readme-stats.vercel.app/api/top-langs/?username=irissuu&layout=compact&bg_color=ff00&title_color=F9A0AF&hide_border=True&include_all_commits=true&count_private=true" width="40%" alt="Github Stats"/>
</a>

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by davevad93** [comment]
Hi @Irissuu ,

From what I've seen in your profile, you have 2 `Java` repos, the thing is those repos are forked and that card from [github-readme-stats](https://github.com/anuraghazra/github-readme-stats) only shows language usage inside your own `public` non-forked repositories. That's why it says `"No languages data"`.

**Reply by Irissuu** [reply]
Thank you, but I tested making my private repositories public, but it didn't work
I also used cont_private=true :(

**Reply by davevad93** [reply]
Hey @Irissuu , that's really odd I've taken a look at your `README` code and everything seems fine. I'm also using that same card from [github-readme-stats](https://github.com/anuraghazra/github-readme-stats) and it works without problems so it can't be an API issue. One question though, when you've changed the visibility of your repos from `private` to `public` did you wait a few hours before making them private again? Because sometimes it takes a while to update the data and maybe that's why no programming language appeared in the card.

I've also tested your same code using my username and it works fine:

<img align="left" src="https://github-readme-stats.vercel.app/api/top-langs/?username=davevad93&layout=compact&bg_color=ff00&title_color=F9A0AF&hide_border=True" width="40%" alt="Github Stats"/>

**Reply by Irissuu** [reply]
I waited, but I also can't make my repositories public yet (because of the other students in my class and the teachers won't let me), my sister uses the same code as me and it worked too, I think it's bugged

**Reply by davevad93** [reply, marked as answer]
Oh I understand @Irissuu , well you could try creating a new repo just for testing, push some code and check if the issue persists because as I said before it's very odd that only you're experiencing this.

**Reply by Irissuu** [reply]
I will, I also contacted a support, tysm.

**Reply by davevad93** [reply]
You're welcome @Irissuu , I hope you'll eventually solve it. Have a nice day!

**Reply by ebndev** [reply]
> I will, I also contacted a support, tysm.

Hey @Irissuu! I see on your profile it's working now - what was the issue?

**Reply by Irissuu** [reply]
@ebndev I'm testing it while talking to support, the code only accepts public repositories, but I would like to present the data both public and private (or just the private repositories would be fine)

**Reply by brito219** [reply]
Hey @Irissuu could you make the private repositories appear? or still the same problem?

**Reply by Irissuu** [reply]
> Hey @Irissuu could you make the private repositories appear? or still the same problem?

still, it's part of the github rules, but I didn't know

**Reply by brito219** [reply]
hey man, i figured it out, let me explain
i actually had to deploy my own instance of the vercel server, for 2 main reasons: the anuraghazra deploy cant possibly hit the rate limiter (5k requests) and the deploy dont have access to the private repos

step by step guide:
1. go to vercel.com and create an account (sign in with github, better)
<img width="1440" alt="68747470733a2f2f66696c65732e636174626f782e6d6f652f7063786b33332e706e67" src="https://github.com/user-attachments/assets/adc7eae4-1ab2-469e-86f3-24273105976a">

2. fork anuraghazra's repo
3. go back to the vercel dashboard and click on "Add New", then create a new project
<img width="1439" alt="68747470733a2f2f66696c65732e636174626f782e6d6f652f336e373666682e706e67" src="https://github.com/user-attachments/assets/1ce4f804-5002-4e9d-93d8-808b6738d1a1">

4. click the Continue with GitHub button, search for the required Git Repository (the forked repo) and import it by clicking the Import button. Alternatively, you can import a Third-Party Git Repository using the Import Third-Party Git Repository -> link at the bottom of the page.
<img width="1440" alt="68747470733a2f2f66696c65732e636174626f782e6d6f652f6d67357030342e706e67" src="https://github.com/user-attachments/assets/7d15da57-cfbb-47d7-b379-cb7a82a71e04">

5. Create a personal access token (PAT) (https://github.com/settings/tokens/new) and enable the repo and user permissions (this allows access to see private repo and user stats).
6. Add the PAT as an environment variable named PAT_1
<img width="783" alt="68747470733a2f2f66696c65732e636174626f782e6d6f652f3079636c696f2e706e67" src="https://github.com/user-attachments/assets/0331693d-51bc-4685-adea-4bc5b6408e61">

7. Click deploy, and you're good to go. change the anuraghazra domain to the new one that you got when deployed 
8. thats it :P

extra: https://www.youtube.com/watch?v=n6d4KHSKqGk&t=107s
to any questions, this video is the answer

**Reply by Irissuu** [reply]
Thank you very much! :D

**Reply by PeteVanBenthuysen** [reply]
Is anyone able to help with this? I did all of this and am still getting "No languages data". Is it because I just uploaded all of my personal projects to repositories on GitHub? I made sure the key had repo and user permissions
